src/components/ui/InfiniteList/index.tsx

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useRef, useEffect, useState, HTMLAttributes } from 'react';
+
+import { StyledInfiniteList } from './Styled';
+import { BaseProps } from '../Base';
+import { Spinner } from '../icons'; // TODO: update icon
+
+export interface InfiniteListProps extends Omit<HTMLAttributes<HTMLUListElement>, 'css'>, BaseProps {
+  /* The elements that will populat the list. These elements will be wrapped in <li> tags, so that can be any element. */
+  items: JSX.Element[];
+  /* A callback function that will make an api call to load the next batch of items. */
+  onLoad: () => Promise<void>;
+  /* Accessibility label of the list */
+  label: string;
+  /* Manages the visibility of the spinner when the API call is resolving */
+  isLoading: boolean;
+}
+
+const InfiniteList = (props: InfiniteListProps) => {
+  const { items, onLoad, label, isLoading } = props;
+
+  const feedRef = useRef<HTMLUListElement>(null);
+  const listEnd = useRef<HTMLDivElement>(null);
+  const currentTopItem = useRef<HTMLLIElement>(null);
+  const firstNew = useRef<HTMLLIElement>(null);
+  const prevLength = useRef(items.length)
+  const newLength = useRef(0);
+
+  const handleLoad = async () => {
+    await onLoad();
+  }
```

Comment:
We aren't awaiting anything since we removed the loading responsibility. We should just be able to call `onLoad` directly in our IntersectionObserver

Reply:
Fixed.